source/custom/nidcpower_service.custom.cpp

```diff
@@ -1,5 +1,51 @@
 #include <nidcpower/nidcpower_service.h>
 
+#include <stdexcept>
+
 namespace nidcpower_grpc {
 
+struct DriverErrorException : std::runtime_error {
+  DriverErrorException(int status) : std::runtime_error("") { status_ = status; }
+  int status_ = 0;
```

Comment:
The `_` suffix belongs on private fields. This is public. Make this a class with a public constructor, public `int status()` method, and private `status_` field.

When making the struct a class, make sure the inheritence is public:
```cpp
class DriverErrorException : public std::runtime_error {
```

Reply:
@jonlugoatx could you possibly elaborate about the `public int status()` method that I have to add? Is it the same as `checkstatus` method that I already have?

Reply:
@jonathan I have converted struct to class. Could you pl review it again?

Reply:
The updated `DriverErrorException` looks in line with Jonathan's comments. Marking resolved.